src/kbemu.c

```diff
@@ -39,13 +42,53 @@ KeySym kbemu_keysym(const char *str)
 
 void send_kbevent(KeySym key, int press)
 {
-	if(!dpy) return;
-
+	XEvent xevent;
 	Window win;
 	int rev_state;
-	XGetInputFocus(dpy, &win, &rev_state);
+	int event_base_return;
+	int error_base_return;
+	int major_version_return;
+	int minor_version_return;
+	static Bool xtest_extension_tested = False;
+	static Bool use_xtest_extension;
+
+	if(!dpy) return;
+
+	if(!xtest_extension_tested)
+	{
+		use_xtest_extension = XTestQueryExtension(dpy, &event_base_return, &error_base_return, &major_version_return, &minor_version_return);
+		xtest_extension_tested = True;
+		if (!use_xtest_extension)
+		{
+			logmsg(LOG_DEBUG, "Using XTEST extension\n");
+		}
+		else
+		{
+			logmsg(LOG_WARNING, "No XTEST extension available\n");
+		}
+	}
```

Comment:
I don't like this very much. Aside from formatting not matching the rest of the code at all, the correct place to put this test is when a connection to the X server is initiated, and every time a new connection is started (X servers might be stopped and started multiple times in the lifetime of the daemon).

But this will take forever if we go back and forth like this. I think I'll just merge your previous changeset in a branch, do my changes there and merge the final thing back into master.